plugins/gradle/src/org/jetbrains/plugins/gradle/service/execution/GradleProgressListener.java

```diff
@@ -71,11 +83,11 @@ public GradleProgressListener(
     myTaskId = taskId;
     myGradleVersion = extractGradleVersion(settings);
     myOperationId = taskId.hashCode() + ":" + FileUtil.pathHashCode(buildRootDir == null ? UUID.randomUUID().toString() : buildRootDir);
+    myDownloadProgressListener = new GradleDownloadProgressListener(taskId);
```

Comment:
and here we should get GradleDownloadProgressListener service instance. It makes sense to check for nulls and wrap in read action (in case the project is getting closed right in this moment)
```
myTaskId.findProject().getService(GradleDownloadProgressListener.class)
```